Source/LibationFileManager/PropertyChangeFilter.cs

```diff
@@ -0,0 +1,335 @@
+
+		[PropertyChangeFilter("MyProperty1")]
+		[PropertyChangeFilter("MyProperty2")]
+		void MyPropertiesChanged(object sender, PropertyChangedEventArgsEx e)
+		{
+			// Only properties whose names match either "MyProperty1"
+			// or "MyProperty2" will fire this event handler.
+		}
+
+	******
+	* OR *
+	******
+	
+		propertyChangeFilter.PropertyChanged +=
+			[PropertyChangeFilter("MyProperty1")]
+			[PropertyChangeFilter("MyProperty2")]
+			(_, _) => 
+			{
+				// Only properties whose names match either "MyProperty1"
+				// or "MyProperty2" will fire this event handler.
+			};
+
+
+		*************************
+		*						*
+		*    Observable Mode	*
+		*						*
+		*************************
+		
+		using var cancellation = propertyChangeFilter.ObservePropertyChanging<int>("MyProperty", MyPropertyChanging);
+		
+        void MyPropertyChanging(int oldValue, int newValue)
+        {
+			// Only the property whose name match
+			// "MyProperty" will fire this method.
+        }
+	
+		//The observer is delisted when cancellation is disposed
+
+	******
+	* OR *
+	******
+	
+		using var cancellation = propertyChangeFilter.ObservePropertyChanged<bool>("MyProperty", s =>
+			{
+				// Only the property whose name match
+				// "MyProperty" will fire this action.
+			});
+		
+		//The observer is delisted when cancellation is disposed
+
+	 */
+
+	#endregion
+
+	public abstract class PropertyChangeFilter
+	{
+		private readonly Dictionary<string, List<Delegate>> propertyChangedActions = new();
+		private readonly Dictionary<string, List<Delegate>> propertyChangingActions = new();
+		
+		private readonly List<(PropertyChangedEventHandlerEx subscriber, PropertyChangedEventHandlerEx wrapper)> changedFilters = new();
+		private readonly List<(PropertyChangingEventHandlerEx subscriber, PropertyChangingEventHandlerEx wrapper)> changingFilters = new();
+
+		public PropertyChangeFilter()
+		{
+			PropertyChanging += Configuration_PropertyChanging;
+			PropertyChanged += Configuration_PropertyChanged;
+		}
+
+		#region Events
+
+		protected void OnPropertyChanged(string propertyName, object newValue)
+			=> _propertyChanged?.Invoke(this, new(propertyName, newValue));
+		protected void OnPropertyChanging(string propertyName, object oldValue, object newValue)
+			=> _propertyChanging?.Invoke(this, new(propertyName, oldValue, newValue));
+
+		private PropertyChangedEventHandlerEx _propertyChanged;
+		private PropertyChangingEventHandlerEx _propertyChanging;
+
+		public event PropertyChangedEventHandlerEx PropertyChanged
+		{
+			add
+			{
+				var attributes = getAttributes<PropertyChangeFilterAttribute>(value.Method);
+
+				if (attributes.Any())
+				{
+					var matches = attributes.Select(a => a.PropertyName).ToArray();
+
+					void filterer(object s, PropertyChangedEventArgsEx e)
+					{
+						if (e.PropertyName.In(matches)) value(s, e);
+					}
+
+					changedFilters.Add((value, filterer));
+
+					_propertyChanged += filterer;
+				}
```

Comment:
If the handler has `PropertyChangeFilterAttribute`s, pass them through the `filterer` delegate and subscribe `filterer` to the event instead of `value`.

Reply:
lol I spent a long time combing through the code to figure out how you did your attribute magic. I should have looked at these comments first. This is literally what I was looking for :foreheadsmack:

Reply:
Yeah, This was originally the first comment to appear. Then I made a minor change and it became outdated, so I re-posted the comment and then it went to the bottom lol.